integration_tests/src/test/kotlin/integration/DumpShardsIT.kt

```diff
@@ -0,0 +1,87 @@
+package integration
+
+import FlankCommand
+import com.google.common.truth.Truth
+import flank.common.isWindows
+import org.junit.Assume
+import org.junit.Test
+import run
+import utils.containsAll
+import utils.loadAndroidDumpShards
+import utils.loadIosDumpShards
+import java.io.File
+
+class DumpShardsIT {
+    private val name = this::class.java.simpleName
+
+    @Test
+    fun `dump shards - android`() {
+        val name = "$name-android"
+        val result = FlankCommand(
+            flankPath = FLANK_JAR_PATH,
+            ymlPath = "$CONFIGS_PATH/dump_shards_android.yml",
+            params = androidRunCommands + "--dump-shards"
+        ).run(
+            workingDirectory = "./",
+            testSuite = name
+        )
+
+        assertExitCode(result, 0)
+
+        val resOutput = result.output.removeUnicode()
+        Truth.assertThat(resOutput).containsMatch(findInCompare(name))
+        assertNoOutcomeSummary(resOutput)
+
+        val matrix = File("android_shards.json").loadAndroidDumpShards()
+
+        Truth.assertThat(matrix.shards.count()).isEqualTo(2)
+
+        Truth.assertThat(matrix.shards.values.flatten()).containsAll(
+            "class com.example.test_app.parametrized.EspressoParametrizedClassParameterizedNamed",
+            "class com.example.test_app.parametrized.EspressoParametrizedClassTestParameterized",
+            "class com.example.test_app.ParameterizedTest",
+            "class com.example.test_app.parametrized.EspressoParametrizedMethodTestJUnitParamsRunner",
+        )
```

Comment:
I like the test checks if there are tests inside shards 👍 We could make one step further and check if there are all tests or we can check just the number. (test count does not change and we check for duplication in another line)